nncf/onnx/graph/model_transformer.py

```diff
@@ -29,25 +27,58 @@
 from nncf.onnx.graph.transformations.commands import ONNXQDQNodeRemovingCommand
 from nncf.common.factory import NNCFGraphFactory
 from nncf.common.graph.model_transformer import ModelTransformer
+from nncf.common.utils.logger import logger
 
 
-# pylint: disable=no-member
 class ONNXModelTransformer(ModelTransformer):
+    """
+    Applies transformations upon ONNX model.
+    ModelTransformer should be created once and be used to apply transformations to the provided model.
+    """
     QUANTIZER_NAME_PREFIX = 'QuantizeLinear_'
     DEQUANTIZER_NAME_PREFIX = 'DequantizeLinear_'
     SCALE_TENSOR_NAME_PREFIX = 'scale_'
     ZERO_POINT_NAME_PREFIX = 'zero_point_'
 
     def __init__(self, model: onnx.ModelProto):
         super().__init__(model)
-        self._model = deepcopy(model)
-        self._nncf_graph = NNCFGraphFactory.create(self._model)
+        self._model = model
+        nncf_graph = NNCFGraphFactory.create(self._model)
+        self.onnx_model_extractor = onnx.utils.Extractor(self._model)
+        for input_node in nncf_graph.get_input_nodes():
+            self._nncf_input_node_next_nodes = {input_node.node_name: nncf_graph.get_next_nodes(input_node)}
+
+    def _get_pre_post_layer_target_edge(self, port_id: int, node_name: str, transform_type: TargetType,
+                                        onnx_graph: ONNXGraph) -> str:
+        """
+        Returns edge name corresponding to node with name equals to node_name, port_id and transforma_type.
+
+        :param port_id: Edge number of port.
+        :param node_name: Node name.
+        :param transform_type: Type of transformation.
+        :param onnx_graph: ONNXGraph
+        :return: Target edge name.
+        """
+        target_edge_name = None
+        if transform_type == TargetType.PRE_LAYER_OPERATION:
+            target_edge_name = onnx_graph.get_node_edge_names(node_name)['input'][port_id]
+        elif transform_type == TargetType.POST_LAYER_OPERATION:
+            if node_name in self._nncf_input_node_next_nodes:  # ADD INPUT NODE CASE
```

Comment:
Why is this case special?

Reply:
Need to have a mapping from NNCF input node in which we apply the quantization scheme to ONNX nodes.

Reply:
Ok. Could you please describe the all manipulations then, because it is not clear, what happens here?

Reply:
I initially create a mapping between NNCF_INPUT_NODES names and consumed their outputs ONNX node names.
Then if the quantizer is applied as POST_LAYER to NNCF_INPUT_NODE I use corresponding ONNX node names to get input edges of the nodes, which are model inputs, on which I need to place Q/DQ.

Reply:
Good. Let's keep this information as the comment in the code.

Reply:
Added description